cli/src/pcluster/templates/cw_dashboard_builder.py

```diff
@@ -219,6 +229,122 @@ def _add_conditional_storage_widgets(
                 widgets_list.append(graph_widget)
         return widgets_list
 
+    def custom_pcluster_metric_filter(self, metric_name, filter_pattern, custom_namespace):
+        """Adding custom metric filter from named tuple."""
+        metric_filter = logs.CfnMetricFilter(
+            scope=self.stack_scope,
+            id=metric_name + " Filter",
+            filter_pattern=filter_pattern,
+            log_group_name=self.cw_log_group_name,
+            metric_transformations=[
+                logs.CfnMetricFilter.MetricTransformationProperty(
+                    metric_namespace=custom_namespace,
+                    metric_name=metric_name,
+                    metric_value="100",
+                    default_value=0,
+                )
+            ],
+        )
+        metric_filter.add_depends_on(self.cw_log_group)
+        return metric_filter
+
+    def add_custom_error_metrics(self):
+        """Create custom error metric filter and outputs to cloudwatch graph."""
+        jobs_not_starting_errors = [
+            _CustomMetricFilter(
+                metric_name="Mismatch between IAM Group",
+                filter_pattern="?UnauthorizedOperation ?AccessDeniedException",
+            ),
+            _CustomMetricFilter(
+                metric_name="AMI larger than root volume",
+                filter_pattern="is smaller than snapshot",
+            ),
+            _CustomMetricFilter(
+                metric_name="Vcpu limit",
+                filter_pattern="VcpuLimitExceeded",
+            ),
+            _CustomMetricFilter(
+                metric_name="Volume Limit",
+                filter_pattern="VolumeLimitExceeded",
+            ),
+            _CustomMetricFilter(
+                metric_name="Node Capacity Insufficient",
+                filter_pattern="InsufficientInstanceCapacity",
+            ),
+        ]
+
+        custom_script_errors = [
+            _CustomMetricFilter(
+                metric_name="Cannot retrieve custom script",
+                filter_pattern="error occurred .* when calling the HeadObject operation Forbidden",
```

Comment:
My bad.
According to https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/FilterAndPatternSyntax.html, it should be:
"error occurred when calling the HeadObject operation"